### ISSUE TITLE
prow-ipv6 and prow-dual-stack templates doesn't have OOT credential provider change

### ISSUE THREAD

**Issue by lzhecheng** [kubernetes-sigs/cluster-api-provider-azure#4394]
/kind bug

[Before submitting an issue, have you checked the [Troubleshooting Guide](https://capz.sigs.k8s.io/topics/troubleshooting.html)?]

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
prow-ipv6 and prow-dual-stack templates doesn't have OOT credential provider change, but the ci-versions ones of these templates do.

**What did you expect to happen:**
prow-ipv6 and prow-dual-stack templates have OOT credential provider change.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by CecileRobertMichon** [comment]
@lzhecheng I just realized the ci-version ipv6 and dual stack templates do have the OOT cred provider changes [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/cluster-template-prow-ci-version-ipv6.yaml#L442) and [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/cluster-template-prow-ci-version-dual-stack.yaml#L227).

If you're asking about the non-ci-version templates, that is already tracked by #4169 and blocked by a Helm chart becoming available to install the credential provider. Non of the templates in the CAPZ repo currently have the OOT credential provider except the ones used to test CI versions due to the need to test it for k/k at HEAD since the install is currently hacky (custom scripts). Once we have a Helm chart available we can install it across all templates via HelmChartProxy (CAAPH).

/close
